### PR TITLE
v0.3.0

### LINKED ISSUE #33: Seed split dataset
For now the split of the dataset is seeded with the same seed everytime
I should:
  1. Set the default seed to None in the parser and take a random if None is given
  2. Save the seed in the model to be able to reproduce the split a posteriori

### LINKED ISSUE #34: Parser default args when restoring training
It would be good to change the behaviour of the parser, so that when restoring training, any command line explicitely specified argument overrides the ones saved in the archive when possible (typically `gibbs_steps` and `learning_rate`)

### PR DESCRIPTION
**Pull request by nbereux**
Do #34

### CODE REVIEW

**Comment by nbereux** [comment]
This pull request was enhanced to do v0.3.0
The breaking change is how the dataset is split. Now by default, a random seed will be used and saved in the model in order to be able to restore the training.

**Comment by nbereux** [comment]
Also the documentation should go in a distinct repository
